From e9a21152f038780593856d59d182319e1684edf5 Mon Sep 17 00:00:00 2001
From: Jason Dere <jdere@hortonworks.com>
Date: Thu, 5 May 2016 12:29:14 -0700
Subject: [PATCH 0760/1164] CDH-44647: HIVE-13620: Merge llap branch work to
 master (committing changes from review feedback)

This only backports part of the change so that the full stacktrace of the exception is carried
from the server to the beeline client.

Change-Id: I133ab66b79165cf135ab03746184282f4728bc67
---
 .../hadoop/hive/ql/parse/TypeCheckProcFactory.java |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/ql/src/java/org/apache/hadoop/hive/ql/parse/TypeCheckProcFactory.java b/ql/src/java/org/apache/hadoop/hive/ql/parse/TypeCheckProcFactory.java
index a036335..07d1c9c 100644
--- a/ql/src/java/org/apache/hadoop/hive/ql/parse/TypeCheckProcFactory.java
+++ b/ql/src/java/org/apache/hadoop/hive/ql/parse/TypeCheckProcFactory.java
@@ -1176,13 +1176,13 @@ public Object process(Node nd, Stack<Node> stack, NodeProcessorCtx procCtx,
         return getXpathOrFuncExprNodeDesc(expr, isFunction, children, ctx);
       } catch (UDFArgumentTypeException e) {
         throw new SemanticException(ErrorMsg.INVALID_ARGUMENT_TYPE.getMsg(expr
-            .getChild(childrenBegin + e.getArgumentId()), e.getMessage()));
+            .getChild(childrenBegin + e.getArgumentId()), e.getMessage()), e);
       } catch (UDFArgumentLengthException e) {
         throw new SemanticException(ErrorMsg.INVALID_ARGUMENT_LENGTH.getMsg(
-            expr, e.getMessage()));
+            expr, e.getMessage()), e);
       } catch (UDFArgumentException e) {
         throw new SemanticException(ErrorMsg.INVALID_ARGUMENT.getMsg(expr, e
-            .getMessage()));
+            .getMessage()), e);
       }
     }
 
-- 
1.7.9.5

